### PR TITLE
Security - kibana_system - add delete privileges for synthetics

### LINKED ISSUE #467: [Elasticsearch] Add necessary `manage` privileges for synthetics indices to the `kibana_system` user to support ILM
Related to https://github.com/elastic/uptime/issues/462

Fleet uses the internal `kibana_system` user to install integration package assets into Elasticsearch in order to support automatic upgrades of packages without end user intervention. To support , we'll need to add extra permissions for this user.

In particular, we must add the `manage` index privilege on all the indices being managed by the ILM policies for Elastic Synthetics, including:

- synthetics-http-*
- synthetics-icmp-*
- synthetics-tcp-*
- synthetics-browser-*
- synthetics-browser.network-*
- synthetics-browser.screenshot-*

Background: https://github.com/elastic/package-spec/issues/293
Example: https://github.com/elastic/elasticsearch/pull/81811

### PR DESCRIPTION
**Pull request by dominiqueclarke**
Resolves https://github.com/elastic/uptime/issues/467

As part of https://github.com/elastic/uptime/issues/462, Uptime will introduce default ILM policies with default delete phases for the `synthetics-http`, `synthetics-tcp`, `synthetics-icmp`, `synthetics-brower`, `synthetics-browser.network` and `synthetics-browser.screenshot` data sets. 

To support this work, the Kibana system user must have extended privileges to delete data streams associated with these data sets.

### CODE REVIEW

**Comment by ywangd** [comment]
ping @elastic/kibana-security

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java [hunk 1]
```diff
@@ -777,7 +777,13 @@ public static RoleDescriptor kibanaSystemRoleDescriptor(String name) {
...
                         "metrics-apm.*-*",
                         "traces-apm-*",
-                        "traces-apm.*-*"
+                        "traces-apm.*-*",
+                        "synthetics-http-*",
+                        "synthetics-icmp-*",
+                        "synthetics-tcp-*",
+                        "synthetics-browser-*",
+                        "synthetics-browser.network-*",
+                        "synthetics-browser.screenshot-*"
```
Comment (azasypkin) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java:786]: __question:__ do we document these patterns anywhere like [we do for APM patterns](https://www.elastic.co/guide/en/apm/guide/8.1/apm-data-streams.html) (`dataset` part in particular, `http/icmp/tcp/browser/browser.network/browser.screenshot`)? I wanted to make sure there are no typos in the names and so on, but couldn't quickly find any references.

**Review by joshdover — APPROVED** [review]
Confirmed this matches the needs of the new ILM policies added in https://github.com/elastic/integrations/pull/2744

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java (hunk 1 above)
Comment (dominiqueclarke) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java:786]: Unfortunately no, we do have this docs ticket where we will address in the future https://github.com/elastic/synthetics/issues/286

**Comment by dominiqueclarke** [comment]
@azasypkin @joshdover I'm not a ES contributor, so I'm hoping one of you can tell me what `
elasticsearch-ci/part-1-fips` is and if I need to worry about the CI failure there.

Also, I have not updated this branch in a week. Do I need to merge upstream similar to Kibana?

**Comment by ywangd** [comment]
@elasticmachine update branch

**Comment by ywangd** [comment]
@dominiqueclarke The above "update branch" should most likely fixes the CI failure.

**Pull request merged**